### PR TITLE
[MRG] add `defaults.conf`; update docs to specify YAML instead of JSON

### PR DESCRIPTION
**Pull request by ctb**
This PR adds a `defaults.conf` and updates the docs in a few ways.

Goals:
* make it so that `showconf` target actually shows _all_ of the config options, not just the ones set in the config file
* provide reasonable (or empty) defaults, provide good error messages when required parameters are not set.

### CODE REVIEW

**Comment by ctb** [comment]
@taylorreiter ready for review and merge (assuming tests pass).

File: doc/01-running-spacegraphcats.md [hunk 1]
```diff
@@ -191,14 +191,14 @@ query, and the reads for the neighborhoods around each query.
 
 ### The `spacegraphcats` script
 
-The `spacegraphcats` script has several targets, in addition to `search` and `searchquick`.
+The `spacegraphcats` script has several targets, in addition to `search`.
 
 * `python -m spacegraphcats twofoo build` will build the catlas
```
Comment (taylorreiter) [inline doc/01-running-spacegraphcats.md]:
```suggestion
* `python -m spacegraphcats run twofoo build` will build the catlas
```

File: doc/01-running-spacegraphcats.md [hunk 2]
```diff
@@ -191,14 +191,14 @@ query, and the reads for the neighborhoods around each query.
...
 * `python -m spacegraphcats twofoo clean` should remove the build targets.
```
Comment (taylorreiter) [inline doc/01-running-spacegraphcats.md]:
```suggestion
* `python -m spacegraphcats run twofoo clean` should remove the build targets.
```

File: doc/01-running-spacegraphcats.md [hunk 3]
```diff
@@ -191,14 +191,14 @@ query, and the reads for the neighborhoods around each query.
...
 * `python -m spacegraphcats twofoo extract_contigs` -- get contigs for search results; see above.
```
Comment (taylorreiter) [inline doc/01-running-spacegraphcats.md]:
```suggestion
* `python -m spacegraphcats run twofoo extract_contigs` -- get contigs for search results; see above.
```

File: doc/01-running-spacegraphcats.md [hunk 4]
```diff
@@ -191,14 +191,14 @@ query, and the reads for the neighborhoods around each query.
...
 * `python -m spacegraphcats twofoo extract_reads` -- get reads for search results; see above.
```
Comment (taylorreiter) [inline doc/01-running-spacegraphcats.md]:
```suggestion
* `python -m spacegraphcats run twofoo extract_reads` -- get reads for search results; see above.
```

File: doc/01-running-spacegraphcats.md [hunk 5]
```diff
@@ -67,55 +67,55 @@ There are three important top-level files.
 
 1. The script `python -m spacegraphcats` uses [snakemake](https://snakemake.readthedocs.io/en/stable/) to run the pipeline.
 
-2. The Snakefile itself is under `conf/Snakefile`, and is not intended to be particularly human readable :). The snakemake run is configured by a JSON config file.
+2. The Snakefile itself is under `conf/Snakefile`, and is not intended to be particularly human readable :). The snakemake run is configured by a [YAML config file](https://www.cloudbees.com/blog/yaml-tutorial-everything-you-need-get-started/).
 
-3. The configuration files are relatively simple, and are (by convention) contained in the `conf/` directory, although you can put them elsewhere.
+3. The configuration files are relatively simple, and are (by convention) contained in the `conf/` directory, although you can put them elsewhere. The filenames should end in either `.yaml` or `.conf`.
```
Comment (taylorreiter) [inline doc/01-running-spacegraphcats.md:72]: `twofoo` doesn't end in `.yaml` or `.conf` when ran from the command line

Comment (taylorreiter) [reply doc/01-running-spacegraphcats.md:72]: oh i see, neither does `empty`. So sgc can just lob off the `yaml` and take just the root name. ok. Not sure if this should be added to docs?

**Review by taylorreiter — COMMENTED** [review]
This is exciting! all of the sgc examples in the `01` doc need `run` added to them. I made suggestions for all of the twofoo ones, but the dory one at the top was out of suggestion reach. it should read:
`python -m spacegraphcats run dory-test search`

Otherwise this looks good. love `python -m spacegraphcats showconf empty`

File: doc/01-running-spacegraphcats.md (hunk 5 above)
Comment (ctb) [reply doc/01-running-spacegraphcats.md:72]: we can leave it as an easter 🥚 for now :)

**Comment by ctb** [comment]
Fixed in 
8ab9691!

**Review by taylorreiter — APPROVED** [review]
LGTM!

**Pull request merged**